lib/platform/github/index.ts

```diff
@@ -1735,3 +1746,82 @@ export async function getVulnerabilityAlerts(): Promise<VulnerabilityAlert[]> {
   }
   return alerts;
 }
+
+async function pushFiles(
+  { branchName, files, message }: CommitFilesConfig,
+  { prevCommitSha }: CommitResult
+): Promise<CommitSha | null> {
+  const additions: FileAddition[] = [];
+  const deletions: FileDeletion[] = [];
+
+  for (const file of files) {
+    if (file.type === 'addition') {
+      additions.push(file);
+    } else {
+      deletions.push(file);
+    }
+  }
+
+  const executableFiles = additions.filter(({ isExecutable }) => isExecutable);
+  if (executableFiles.length) {
+    logger.warn(
+      { branchName, executableFiles: executableFiles.map(({ path }) => path) },
+      'Platform-native commit: found executable files'
+    );
+  }
+
+  const fileChanges: GithubGraphqlFileChanges = {
+    additions: additions.map(({ path, contents }) => ({
+      path,
+      contents: Buffer.from(contents).toString('base64'),
```

Comment:
Does it work when contents is already a buffer?

Reply:
✅ `Buffer.from(Buffer.from('foobar')).toString() === 'foobar'`